packages/worker-capacity-manager/src/sqs.ts

```diff
@@ -0,0 +1,28 @@
+import { GetQueueAttributesCommand, SQSClient } from '@aws-sdk/client-sqs';
+
+export const getSQSQueueLengthIncludingInvisible = async (
+	sqsClient: SQSClient,
+	queueUrl: string,
+): Promise<number> => {
+	try {
+		const command = new GetQueueAttributesCommand({
+			QueueUrl: queueUrl,
+			AttributeNames: [
+				'ApproximateNumberOfMessages',
+				'ApproximateNumberOfMessagesNotVisible',
+			],
+		});
+		const response = await sqsClient.send(command);
+		const attributes = response.Attributes;
+		if (attributes) {
+			return (
+				Number(attributes.ApproximateNumberOfMessages) +
+				Number(attributes.ApproximateNumberOfMessagesNotVisible)
+			);
+		}
+	} catch (error) {
+		console.log("Couldn't get queue length", error);
+		throw error;
+	}
+	throw new Error('unable to retrieve sqs message count attributes');
+};
```

Comment:
When can this throw occur? Is it only when `attributes` is undefined? If that's the case, could it not be as an else at the point where we check the `attributes`?

Reply:
Intellij gave me bad advice! I did have it this way originally, changed it back!